### ISSUE TITLE
Numba dependency incorrect

### ISSUE THREAD

**Issue by mdekstrand** [lenskit/lkpy#129]
We are seeing run failures in certain Google Collab environments - a `numba` typing failure.

- environment [requirements.txt](https://github.com/lenskit/lkpy/files/4003007/collab-requirements.txt)

Reproducer code:

```
from lenskit.datasets import MovieLens
from lenskit.algorithms.user_knn import UserUser
from lenskit.algorithms import Recommender

ml = MovieLens('ml-latest-small')
uu = UserUser(30)
uu = Recommender.adapt(uu)
uu.fit(ml.ratings)
```

Removing the `Recommender.adapt` call removes the error.

**Comment by mdekstrand** [comment]
This is a Numba compatibility problem. Upgrading to Numba 0.46 fixes it.

**Comment by mdekstrand** [comment]
Fixed in #130 and ready to release.
